mycpp/gc_builtins.cc

```diff
@@ -255,3 +257,29 @@ bool str_equals0(const char* c_string, Str* s) {
     return false;
   }
 }
+
+int hash(Str* s) {
+  std::hash<void*> h;
+  return h(static_cast<void*>(s));
```

Comment:
this could potentially just be
```suggestion
  return reinterpret_cast<int>(s);
```

Reply:
To match Python, we should hash the CONTENTS of the string

The use of hash() in `comp_ui.py` is weird and I wonder if it could be improved (open to suggestions)

But I think the shortest solution is to hash the sting for now

Reply:
oh you're right... for some reason i thought `hash(str) == id(str)` in python. i'll update this